### DISCUSSION TITLE
How to get the params in the links function?

### DISCUSSION THREAD

**Discussion by ric980** [remix-run/remix#9334] [Q&A]
Hello,

I need to setup canonical and alternate meta link tags in the head of my website pages. But to do that I need to get the url path so that I can make decisions what is canonical and what is alternate. But the LinksFunction takes no arguments, like the MetaFunction. I need to make these decisions server side.

Example:

url = https://example.com/products/$id

Now I need to setup links like these:

```
<link rel="alternate" href="https://example.com/en/products/${params.id} hrefLang="en">
<link rel="alternate" href="https://example.com/bg/products/${params.id} hrefLang="bg">
<link rel="alternate" href="https://example.com/de/products/${params.id} hrefLang="de">
```

How do I do that?

I know i18n is not a focus for Remix, but people do develop multi-language sites and have problems :) 

Thanks.

**Comment by kiliman** [comment]
Yes, the `links` export doesn't have access to the loader data.

However, you can use the `meta` function to return links.

https://remix.run/docs/en/main/route/meta

```ts
export const meta = ({ data }) => [
  {
    tagName: 'link',
    rel: 'alternate',
    href: `https://example.com/en/products/${data.id}`,
    hrefLang: 'en',
  },
  // ...
]
```

**Comment by ric980** [comment]
Thank you so much. That works fine.
